src/legacy/core_plugins/vis_default_editor/public/components/sidebar/state/index.ts

```diff
@@ -17,42 +17,31 @@
  * under the License.
  */
 
-import { useEffect, useReducer, useCallback } from 'react';
-import { isEqual } from 'lodash';
+import { useReducer, useCallback } from 'react';
 
-import { Vis, VisState, VisParams } from 'src/legacy/core_plugins/visualizations/public';
-import { editorStateReducer, initEditorState } from './reducers';
+import { Vis } from 'src/legacy/core_plugins/visualizations/public';
+import { editorStateReducer, EditorVisState, initEditorState } from './reducers';
 import { EditorStateActionTypes } from './constants';
-import { EditorAction, updateStateParams } from './actions';
+import { EditorAction } from './actions';
 
 export * from './editor_form_state';
 export * from './actions';
 
-export function useEditorReducer(vis: Vis): [VisState, React.Dispatch<EditorAction>] {
+export function useEditorReducer(
+  vis: Vis,
+  dirtyStateChange: any
+): [EditorVisState, React.Dispatch<EditorAction>] {
   const [state, dispatch] = useReducer(editorStateReducer, vis, initEditorState);
 
-  useEffect(() => {
-    const handleVisUpdate = (params: VisParams) => {
-      if (!isEqual(params, state.params)) {
-        dispatch(updateStateParams(params));
-      }
-    };
-
-    // fires when visualization state changes, and we need to copy changes to editorState
-    vis.on('updateEditorStateParams', handleVisUpdate);
```

Comment:
This listener was registered to address changes from a vis, there is still a case exists in 
`src/legacy/core_plugins/vis_type_table/public/vis_controller.ts`

![image](https://user-images.githubusercontent.com/31325372/76937362-b7586d00-6905-11ea-94be-fdaa6ed9d01d.png)

I'm not sure whether this is still needed (could we change editor state through a visualization ?),
but this should be re-checked and removed if not, or make it compatible

Reply:
I did some investigation on this and AFAICT it is not needed for data table vis. Removed the code and so far have not caught any issues with it.